agent/consul/internal_endpoint.go

```diff
@@ -185,7 +185,7 @@ func (m *Internal) ServiceDump(args *structs.ServiceDumpRequest, reply *structs.
 			for _, p := range listedPeerings {
 				index, importedNodes, err := state.ServiceDump(ws, args.ServiceKind, args.UseServiceKind, &args.EnterpriseMeta, p.Name)
 				if err != nil {
-					return err
+					return fmt.Errorf("could not get a service dump for peer's %q nodes: %w", p.Name, err)
```

Comment:
Same as above, could be shortened:
`"could not get a service dump for peer %q: %w"`